getting_serious/puzzle_29.R

```diff
@@ -0,0 +1,47 @@
+## Ten people, two rooms, states are {0, 1, 2, ..., 10} (number of people in other room)
+states <- seq(0, 10)
+
+## Transition matrix P
+## P[i, j] is the probability of transitioning from state i to j
+P <- matrix(0, length(states), length(states))
+rownames(P) <- states
+colnames(P) <- states
+for(i in seq_along(states)[seq(2, length(states) - 1)]) {
+    P[i, i-1] <- (i-1) / max(states)  # Person in other room selected, state decreases
+    P[i, i+1] <- (max(states) - i + 1) / max(states)
+}
+
+## Edge cases: when one room is full,
+## one person will move to the empty room (with probability one)
+P[1, 2] <- 1
+P[length(states), length(states) - 1] <- 1
+
+all(rowSums(P) == 1)  # Careful with float comparisons
+
+## Initial distribution -- initial, the state is equally likely to be 0 or 1
+pi <- c(0.5, 0.5, rep(0, length(states) - 2))
+
+t(pi) %*% P  # Distribution after one period
+t(pi) %*% P %*% P  # Distribution after two periods
+t(pi) %*% P %*% P %*% P  # After three...
+
+time_max <- 100  # What is the distribution after time_max periods?
+curr_pi <- pi
```

Comment:
Does the `i` in the variable name `pi` stand for _initial_? If so, `curr_pi` should rather be `curr_p`, or just `p`.

Should the `pi` rather be `t(pi)`, for consistency with the paragraph above? I tried both, and R seemed to accept both variants, producing the same results.

Reply:
It's the Greek pi, which is very commonly used notation in Markov chain problems when solving for the stationary distribution

See https://en.wikipedia.org/wiki/Markov_chain or http://www.stat.yale.edu/~pollard/Courses/251.spring2013/Handouts/Chang-MarkovChains.pdf for example

![Screenshot from 2022-04-26 22-11-42](https://user-images.githubusercontent.com/1395971/165445158-ebb7a2ee-0c7d-4835-83bb-2a16eb3a3109.png)